source/core_mqtt_serializer.c

```diff
@@ -519,7 +519,7 @@ static uint8_t * encodeString( uint8_t * pDestination,
     pBuffer++;
 
     /* Copy the string into pBuffer. */
-    if( pSourceBuffer != NULL )
+    if( ( pSourceBuffer != NULL ) && ( sourceLength != 0 ) )
```

Comment:
Why does a pull request removing mallocCanFail from proofs make a modification to the project code?

Reply:
This change is unnecessary, I reverted it.